word-level-pos-tokens-recogs-style-decoder-loop.rasp

```diff
@@ -934,14 +990,15 @@ no_pp_np_mask = 1 - aggregate((pp_one_after_mask and np_prop_diag_mask) or (pp_t
 nps_without_pp_prefix_indices = selector_width(select(NOUN_MASK*no_pp_np_mask, 1, ==) and select(indices, indices, <=))*NOUN_MASK*no_pp_np_mask;
 
 left_idx = aggregate(select(indices, left_idx_in_nps_zero_based, ==), input_indices_sorted);
-right_idx = aggregate(select(nps_without_pp_prefix_indices, after_intro_idx, ==), indices);
+right_idx = aggregate(select(nps_without_pp_prefix_indices, after_intro_idx+cp_in_output_count, ==), indices);
 
 # points to 2nd verb for xcomp for v_inf_taking_v_inf
 right_idx = aggregate(select(indices, (nv_in_output_count-1), ==), input_indices_sorted) if (template_name == "v_inf_taking" and after_intro_idx == 2) else right_idx;
 
 
 # points to 1st noun for 2nd v_inf agent in v_inf_taking_v_inf
-right_idx = aggregate(select(indices, 0, ==), input_indices_sorted) if (template_name == "v_inf_taking" and after_intro_idx == 3) else right_idx;
+nps_without_pp_prefix_indices_after_cp_one_based = (selector_width(select(CP_SEP_AFTER_ALL_MASK*NOUN_MASK*no_pp_np_mask, 1, ==) and select(indices, indices, <=)))*CP_SEP_AFTER_ALL_MASK*NOUN_MASK*no_pp_np_mask;
```

Comment:
Can this be simplified to avoid creating a new mask or to multiply by CP_SEP_AFTER_ALL_MASK here? Reuse the mask from the previous right_index?

If this is just getting the first non-"pp np" noun after all complement phrases, isn't that just an offset of cp_in_input_count?